CppcheckTargets.cmake

```diff
@@ -27,19 +27,29 @@ endif()
 set(__add_cppcheck YES)
 
 if(NOT CPPCHECK_FOUND)
-	find_package(cppcheck QUIET)
+	find_package(cppcheck)
 endif()
 
-if(CPPCHECK_FOUND)
-	if(NOT TARGET all_cppcheck)
-		add_custom_target(all_cppcheck)
-		set_target_properties(all_cppcheck PROPERTIES EXCLUDE_FROM_ALL TRUE)
-	endif()
+if(NOT CPPCHECK_FOUND)
+  add_custom_target(cppcheck
+    COMMENT "cppcheck executable not found")
+  set_target_properties(cppcheck PROPERTIES EXCLUDE_FROM_ALL TRUE)
+elseif(CPPCHECK_VERSION VERSION_LESS 1.53.0)
+  add_custom_target(cppcheck
+    COMMENT "Need at least cppcheck 1.53, found ${CPPCHECK_VERSION}")
+  set_target_properties(cppcheck PROPERTIES EXCLUDE_FROM_ALL TRUE)
+  set(CPPCHECK_FOUND)
+endif()
+
+if(NOT TARGET cppcheck)
+  add_custom_target(cppcheck)
 endif()
 
 function(add_cppcheck_sources _targetname)
 	if(CPPCHECK_FOUND)
-		set(_cppcheck_args)
+		set(_cppcheck_args --force -I ${CMAKE_SOURCE_DIR}
```

Comment:
Any particular reason why those args are being forced? I can imagine that the `error` one is for a proper exit to the build, and the `inline-suppr` and `unmatchedsuppression` are for code-embedded suppressions and compatibility with versions that don't support the same suppressions. The automatic include path seems a bit presumptuous, though maybe not. I don't remember offhand what `--force` does.